worlds/rabi_ribi/client/client.py

```diff
@@ -715,6 +724,7 @@ async def rabi_ribi_watcher(ctx: RabiRibiContext):
 
             if ctx.in_state_where_can_give_items():
                 await ctx.give_item()
+                await ctx.add_remote_eggs()
```

Comment:
I think it might be more readable to move the `is_item_queued()` check to here, e.g.:

```python
if ctx.in_state_where_can_give_items():
    if ctx.is_item_queued():
        await ctx.give_item()
    if ctx.is_egg_queued():
        await ctx.add_remote_eggs()
```